bundles/org.openhab.io.transport.modbus/src/main/java/org/openhab/io/transport/modbus/internal/ModbusManagerImpl.java

```diff
@@ -125,15 +135,53 @@ public void accept(AggregateStopWatch timer, T task, ModbusSlaveConnection conne
         // Compare request and response transaction ID. NOTE: ModbusTransaction.getTransactionID() is static and
         // not safe to use
         if ((response.getTransactionID() != libRequest.getTransactionID()) && !response.isHeadless()) {
-            logger.warn(
-                    "Transaction id of the response ({}) does not match request ({}) {}. Endpoint {}. Ignoring response. [operation ID {}]",
-                    response.getTransactionID(), libRequest.getTransactionID(), task.getRequest(), task.getEndpoint(),
-                    operationId);
             throw new ModbusUnexpectedTransactionIdException(libRequest.getTransactionID(),
                     response.getTransactionID());
         }
     }
 
+    /**
+     * Check that function code of the response and request match
+     *
+     * @param response
+     * @param libRequest
+     * @param task
+     * @param operationId
+     * @throws ModbusUnexpectedTransactionIdException
```

Comment:
If you are going to include javadocs at least fill them out. Or get rid of the javadocs....

Reply:
You are right, added explanations.